### DISCUSSION TITLE
How to install the package for reviewing Chinese?

### DISCUSSION THREAD

**Discussion by Pulongon** [asreview/asreview#1426] [Q&A] [answered]
Hello Everybody!

I am a student of Taiwan.
And I need to read Chinese when I use ASReview Lab.
How can I do?

It seems that I should install the package: **multilingual feature extractor**, right?

However, I failed to install the package for many times!
Can somone teach me how to install it correctly?

I also thank Jelle Teijema. He wrote a model for me.
However, I have tried the steps that he provided to me:

1. pip install -U sentence-transformers

1. pip install -U https://github.com/asreview/asreview-multilingual-feature-extractor/archive/master.zip

I copied the code above and executed it, but it failed. I have the error information below.

![ASReview](https://user-images.githubusercontent.com/95769300/236638704-426b3736-b63f-4955-ba2d-e8c1f29026f9.png)

**Comment by rohitgarud** [comment]
Hi @Pulongon,
Did you try 

```
pip install git+https://github.com/asreview/asreview-multilingual-feature-extractor.git
``` 

as mentioned in the extension doc? I was able to install it using this command

**Reply by Pulongon** [reply]
Hi! @rohitgarud !
Thank for your suggestion!

Now I try the command, and I get the message below.
Can you help me? Do I need to install "git"?(What is git?)

![git](https://user-images.githubusercontent.com/95769300/236659755-a6cea35e-42bc-4902-bab5-9bbd8401f0b8.png)

**Reply by rohitgarud** [reply]
@Pulongon ,
Install git from https://git-scm.com/download/win using the standalone installer and make sure you add that to your system PATH. To check if git is properly installed and is added to the PATH, open the command prompt and run `git`, if it gives instructions for using git means it is installed correctly. If it gives an error like 'git' is not recognized, add to PATH using https://stackoverflow.com/questions/26620312/git-installing-git-in-path-with-github-client-for-windows

**Reply by Pulongon** [reply]
Dear @rohitgarud
I haved installed git and run the program again.
It still failed......

But I thank for your help!


![git](https://user-images.githubusercontent.com/95769300/236667481-6d8a8523-ee0b-4548-a461-c916db5decad.png)

**Reply by rohitgarud** [reply]
Oh.. then the last option is to copy the entire error and ask ChatGPT what's wrong 😄

**Reply by jteijema** [reply]
The problem is with SKLearn! We should update the repo to use scikit-learn as a dependency instead of sklearn (or consider removing it altogether).

**Comment by Robdboer** [comment]
Hi @Pulongon , thank you for your message. I ran into a similar problem this morning while installing the package. The issue seems to be related to the 'charmap' codec not being able to decode a specific character in the setup.py file of sklearn. What fixed it for me was setting the environment variable PYTHONUTF8 to 1, which will make Python use the UTF-8 encoding by default. 
Try executing 'set PYTHONUTF8=1' before installing the multilingual feature extractor package. Please let me know if it worked!
Best regards,
Rob

**Reply by Robdboer** [reply, marked as answer]
![image](https://user-images.githubusercontent.com/90698352/236776191-e9f8e654-aadb-4519-aa49-813c47ad3a4b.png)

**Reply by Robdboer** [reply]
Please note that the multilingual sentence transformer won’t work in combination with the naive bayes classifier

**Reply by Pulongon** [reply]
To @Robdboer 
My Dear Friend!
You are wonderful! I follow the steps that you teach me, and I success!

And I also thank @rohitgarud and @jteijema for your help!
Last night, @jteijema still thought about my problems and gave me some suggestions.

Thank you all! I will make the most of ASReview Lab. It is a great invention!

Sincerely,
Chih-lin, Liu

Taiwan

![Success](https://user-images.githubusercontent.com/95769300/236809912-418c1e5c-24ba-4151-86d6-0441a1f21d5f.png)
![Success 2](https://user-images.githubusercontent.com/95769300/236809921-e93468b3-3bfb-4707-aa96-63a5a6095bd0.png)
